lib/functions/aeminitiateupload.js

```diff
@@ -34,6 +35,7 @@ const { MIMETYPE } = require("../constants");
  * @property {Number} [retryInterval=100] time between retries, used by exponential backoff (ms)
  * @property {Boolean} [retryEnabled=true] retry on failure enabled
  * @property {Boolean} [retryAllErrors=false] whether or not to retry on all http error codes or just >=500
+ * @property {Boolean} [supportSpecialCharacters=false] If true, adds support for special characters # and & by url encoding the AEM initiateUpload and completeUpload paths, and adjusting checks in FailUnsupportedAssets. If false, no special logic will be performed.
```

Comment:
```suggestion
 * @property {Boolean} [supportSpecialCharacters=false] If true, adds support for special characters # and & by encoding the AEM initiateUpload and completeUpload paths used for URLs, and adjusting checks in FailUnsupportedAssets. If false, no special logic will be performed. Defaults to false.
```
(I always like to explicitly see what the default is)

Reply:
When will `supportSpecialCharacters` be `true`?  Does that depend on the AEM release version?

Reply:
The thinking is `supportSpecialCharacters` would only be enabled if explicitly requested by a client (e.g. Asset Link). This would be done by adding the `supportSpecialCharacters` flag when calling `uploadFiles()`.

For example:

```
const aemUpload = new AEMUpload();

aemUpload.uploadFiles({
  requestOptions,
  headers,
  uploadFiles: [
    {
      fileUrl: assetDestinationPath,
      filePath: assestLocalPath,
      fileSize: stats.size,
    },
  ],
  supportSpecialCharacters: true,
});
```